nest/proto.c

```diff
@@ -83,18 +98,85 @@ static void
 proto_log_state_change(struct proto *p)
 {
   if (p->debug & D_STATES)
+  {
+    char *name = proto_state_name(p);
+    if (name != p->last_state_name_announced)
     {
-      char *name = proto_state_name(p);
-      if (name != p->last_state_name_announced)
-	{
-	  p->last_state_name_announced = name;
-	  PD(p, "State changed to %s", proto_state_name(p));
-	}
-    }
-  else
+	    p->last_state_name_announced = name;
+	    PD(p, "State changed to %s", proto_state_name(p));
+  	  notify_bgp_stat_daemon(p);
+	  }
+  }
+  else{
     p->last_state_name_announced = NULL;
+  }
+}
+
+/**
+ * Function sends a UDP message to the BGP stat daemon notifying the state of the router
+ * This function will be called whenever the router state changes
+ * @proto: pointer to struct proto
+ */
+static void
+notify_bgp_stat_daemon(struct proto *P)
+{
+  if (!P)
+  {
+    return;
+  }
+
+  // Checks if it is a BGP protocol to send data to the BGP daemon
+  if (P->cf->protocol == &proto_bgp && P->proto)
+  {
+    struct bgp_proto * p = (struct bgp_proto *)P;
+    const char * state = bgp_state_dsc(p);
+    char * name = P->name;
+    char message[1023 + 1];
+    struct bgp_config * cf = p->cf;
+    time_t time_of_change = time(NULL);
+    char buf[127 + 1];
+    struct tm * ts = gmtime(&time_of_change);
+    strftime(buf, sizeof(buf) - 1, "%Y-%m-%d-%H-%M-%S", ts);
+    bsnprintf(message, 1023, "%s$%I$%s$%s", name, cf->remote_ip, state, buf);
```

Comment:
They are guaranteed to be "short". These are hardcoded protocol identifiers used in very few places in the code. The only names that are currently in use are `Kernel`, `Device`, `Static`, `RIP`, `RAdv`, `Pipe`, `OSPF`, `BGP`, `BFD`, and `Direct`.

There is not specific need to transfer the name actually...